src/htmx.js

```diff
@@ -1929,18 +1929,18 @@ return (function () {
         function addHxOnEventHandler(elt, eventName, code) {
             var nodeData = getInternalData(elt);
             nodeData.onHandlers = [];
-            var func = new Function("event", code + "; return;");
             var listener = function (e) {
-                return func.call(elt, e);
+                return maybeEval(elt, function() {
+                    new Function("event", code).call(elt, e);
```

Comment:
seems like this would recompile every time the event is called, should maybe be left up above?

htmx doesn't dynamically reevaluate stuff for the most part, you have to explicitly call `htmx.process()` on a node again to update behavior

Reply:
thank you for the tests, by the way, very well done!

Reply:
Re: function eval, so I wasn't too sure about this, it seemed to me that the caching of the function constructor was a premature optimization that may become a source of surprise if you were to dynamically change the content of the attribute. ie. the behaviour would be inconsistent with regular html `on` event attrs. It also seemed to me to be a more consistent use of `maybeEval` to interpret the code at the point of call, which is how `maybeEval` is used elsewhere. 
Does `new Function` actually pre-compile anyway?
Does it actually have a measurable speed-up by caching the `new Function` construction?
Personally I'd avoid the caching of the function construction, unless there is a noticable performance issue, and caching noticably improves upon that.

Reply:
Hmmm, just thinking about this, I guess the dynamic ability is a moot point for the hx-on attr anyway as it encodes the event name, and that can't be changed without the `htmx.process()` call to update.

Although, I'd expect `new Function` to never be called if `allowEval` was false, so maybe if there is an advantage to caching the `new Function` call, maybe do so on first trigger of the event.

I'm happy to adjust the PR if a consensus on approach is reached.

Reply:
> if you were to dynamically change the content of the attribute

If you dynamically change the content of the attribute without calling `htmx.process` you're going to break a lot of stuff anyway. You're right that it's not consistent with html `on` events but it *is* consistent with the current behavior `hx-on`.

Reply:
I've updated the PR to create and cache the function on first trigger.

Reply:
I looked at your update, thought "there's no way that's the simplest way to accomplish that," tried to improve it, failed, and concluded that this is the simplest way to accomplish that.